cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs

```diff
@@ -4137,9 +4137,19 @@ public DateTimeOffset GetDpsProcessingDate()
         /// <returns></returns>
         public DateTimeOffset GetDpsUploadedDate()
         {
+            // new calculation for DPS processing Date
+            // Check DocumentStatus = Uplaoded 100,000,010
+            // Document Business are not equal to remedial
+            // Check for Compliance date < Uplaoded Date
+            // Check DPS class is 9-General
+            // DFTDP # 977 Change made on 05/02/2024
+
             var mostRecentRecord = dynamicsContext.bcgov_documenturls
-                .Where(x => x.dfp_uploadeddate != null)
-                .OrderByDescending(i => i.dfp_uploadeddate)
+                .Where(x => x.dfp_uploadeddate != null 
+                && x.dfp_submittalstatus == 100000010 // Document status Uplaoded
+                && x.dfp_dpsclass.Contains("9 - General") // Check for 9 - General
+                && x.dfp_DocumentTypeID.dfp_businessarea != 100000001) // Does not contain remedial
+                .OrderBy(i => i.dfp_uploadeddate)
```

Comment:
I expect this should be OrderByDescending - do we want the most recent date or the oldest?

Reply:
Yes the criteria is to order by oldest uploaded date.